samples/emoji-search/presenter/src/commonTest/kotlin/com/example/redwood/emojisearch/presenter/EmojiSearchWidgets.kt

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (C) 2023 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.redwood.emojisearch.presenter
+
+import app.cash.redwood.LayoutModifier
+import app.cash.redwood.RedwoodCodegenApi
+import app.cash.redwood.layout.widget.Column
+import app.cash.redwood.layout.widget.RedwoodLayoutWidgetFactory
+import app.cash.redwood.layout.widget.Row
+import app.cash.redwood.treehouse.lazylayout.widget.LazyColumn
+import app.cash.redwood.treehouse.lazylayout.widget.RedwoodTreehouseLazyLayoutWidgetFactory
+import com.example.redwood.emojisearch.widget.EmojiSearchWidgetFactories
+import com.example.redwood.emojisearch.widget.EmojiSearchWidgetFactory
+import com.example.redwood.emojisearch.widget.Image
+import com.example.redwood.emojisearch.widget.Text
+import com.example.redwood.emojisearch.widget.TextInput
+import example.values.TextFieldState
+
+// TODO(swankjesse): GENERATE THIS.
+
+@OptIn(RedwoodCodegenApi::class)
+fun EmojiSearchTester(): RedwoodTester {
+  return RedwoodTester(
+    EmojiSearchWidgetFactories(
+      EmojiSearch = MutableEmojiSearchWidgetFactory,
+      RedwoodLayout = MutableRedwoodLayoutWidgetFactory,
+      RedwoodTreehouseLazyLayout = MutableRedwoodTreehouseLazyLayoutWidgetFactory,
+    ),
+  )
+}
+
+@RedwoodCodegenApi
+private object MutableEmojiSearchWidgetFactory : EmojiSearchWidgetFactory<MutableWidget> {
+  override fun TextInput(): TextInput<MutableWidget> = MutableTextInput()
+
+  override fun Text(): Text<MutableWidget> = MutableText()
+
+  override fun Image(): Image<MutableWidget> = MutableImage()
+}
+
+@RedwoodCodegenApi
+private class MutableTextInput : TextInput<MutableWidget>, MutableWidget {
+  override val value: MutableWidget
+    get() = this
+
+  override var layoutModifiers: LayoutModifier = LayoutModifier
+
+  var state: TextFieldState? = null
+  var hint: String? = null
```

Comment:
I didn’t differentiate between never-set and null, for properties that can be null. I think that’s fine.

Reply:
Because the source of these properties is function arguments from the composable, there's no such thing as never-set properties.